library/ssl_tls.c

```diff
@@ -2063,44 +2063,64 @@ static int ssl_decrypt_buf( mbedtls_ssl_context *ssl )
  */
 static int ssl_compress_buf( mbedtls_ssl_context *ssl )
 {
-    int ret;
-    unsigned char *msg_post = ssl->out_msg;
```

Comment:
why did you remove `msg_post`, `len_pre` and `msg_pre`?

Reply:
(Applies also to your other comment on the same matter...) Because each were only set to a single variable, I found they were just getting in the way of my understanding what the code was doing.

```
-    unsigned char *msg_post = ssl->out.msg;
-    size_t len_pre = ssl->out.msglen;
-    unsigned char *msg_pre = ssl->compress_buf;
```

In the new code, we hold state in a loop in the original vars.  If we kept the temp vars, I would have to modify them in the loop and then update the originals outside with the temp vars.  It's not bringing anything to the party.